docs/usage/plugins.md

```diff
@@ -0,0 +1,113 @@
+!!! warning "Experimental feature"
+    Plugins support is experimental and is subject to change in minor releases.
+    Developing plugins is not recommended until the feature becomes stable.
+
+Pydantic allows users to create plugins that can be used to extend the functionality of the library.
+
+Plugins are installed via Python entry points. You can read more about entry points in the
+[Entry points specification](https://packaging.python.org/specifications/entry-points/) from the
+Python Packaging Authority.
+
+In case you have a project called `my-pydantic-plugin`, you can create a plugin by adding the following
+to your `pyproject.toml`:
+
+```toml
+[project.entry-points.pydantic]
+my_plugin = "my_pydantic_plugin:plugin"
+```
+
+The entry point group is `pydantic`, `my_plugin` is the name of the plugin, `my_pydantic_plugin` is the module to load plugin object from, and `plugin` is the object name to load.
+
+Plugins are loaded in the order they are found, and the order they are found is not guaranteed.
+
+As a user, you can modify the behavior of the plugin in a `BaseModel` using the `plugin_settings` [Model Config](../usage/model_config.md) argument or
+class keyword argument. This argument takes a dictionary of settings that will be passed to all plugins as is.
+The plugin can then use these settings to modify its behavior. It is recommended for plugins to separate their settings
+into their own dedicates keys in a plugin specific key in the `plugin_settings` dictionary.
+
+```py test="skip"
+from pydantic import BaseModel
+
+
+class Foo(BaseModel, plugin_settings={'my-plugin': {'observe': 'all'}}):
+    ...
+```
+
+## Build a plugin
+
+??? api "API Documentation"
+    [`pydantic.plugin.plugin`][pydantic.plugin.plugin]<br>
+
+Pydantic has an API for creating plugins. The API is exposed via the `pydantic.plugin` module.
```

Comment:
There is a similar sentence above

> Pydantic allows users to create plugins that can be used to extend the functionality of the library.

This one feels like a first sentence of a document but we already have one like that. It's stylistic and just felt like that. Feel free to ignore.

Reply:
feel free to tweak.